packages/entity-server/src/routes/hierarchy/middleware/attachEntityContext.ts

```diff
@@ -13,8 +13,12 @@ const throwNoAccessError = (entityCodes: string[]) => {
   throw new PermissionsError(`No access to requested entities: ${entityCodes}`);
 };
 
-const userCanAccessEntity = (entity: EntityType, allowedCountries: string[]) =>
-  entity.isProject() ||
+const userCanAccessEntity = (
+  entity: EntityType,
+  allowedCountries: string[],
+  rootEntity: EntityType,
+) =>
+  (entity.isProject() && entity.code === rootEntity.code) ||
```

Comment:
Bug fix here! Previously this check would pass if you requested for multiple project entities. This is impossible, since there can only be one project entity in the hierarchy, so checking here to ensure the project entity is the root hierarchy entity